incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -45,6 +45,44 @@ var _ = Describe("Secret", func() {
 		cleanupObjects(ctx)
 	})
 
+	// Exceptions have not been implemented, so we are ignoring the following three tests for now.
+	// They should be enabled once exception is done.
+	PIt("should propagate object only to selected namespace using treeSelect", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		// Create a secret that does NOT propogate to bazName
+		a := map[string]string{"propagate.hnc.x-k8s.io/treeSelect": "!" + bazName}
+		makeObjectWithAnnotation(ctx, "Secret", fooName, "fooSecret", a)
+
+		Eventually(hasObject(ctx, "Secret", barName, "fooSecret")).Should(BeTrue())
+		Consistently(hasObject(ctx, "Secret", bazName, "fooSecret")).Should(BeFalse())
+	})
+
+	PIt("should propagate object only to selected namespace using select", func() {
+		setParent(ctx, barName, fooName)
+		setParent(ctx, bazName, fooName)
+
+		// Create a secret that does NOT propogate to bazName
+		a := map[string]string{"propagate.hnc.x-k8s.io/select": "propagate.hnc.x-k8s.io/!" + bazName}
```

Comment:
The "not" goes at the beginning of the key: `!propagate.hnc.x-k8s.io/baz-123456`.

Reply:
done